components/user/UserProfile.js

```diff
@@ -76,15 +76,19 @@ function UserProfile({ BASE_URL, data }) {
       {!qrShow && (
         <div className="flex flex-wrap justify-center">
           {data.tags?.length > 0 &&
-            data.tags.map((tag) => (
-              <Tag
-                name={tag}
-                key={tag.toLowerCase()}
-                onClick={() =>
-                  router.push(`/search?keyword=${tag.toLowerCase()}`)
-                }
-              />
-            ))}
+            data.tags.map((tag, index) => {
```

Comment:
Maybe a `flatMap` would be better here, so items can be removed?

Reply:
There is no need for flatting the array , we only need to remove falsy values.
We can also filter the items before mapping , but that will result into twice iteration over the tags list.
That's why I have choosen that approach.